City-of-Helsinki/helsinki-paatokset
# UHF-10016: Sonarcloud code quality fixes

# [UHF-10016](https://helsinkisolutionoffice.atlassian.net/browse/UHF-10016)

## What was done

* Fix security and code quality issues found by Sonarcloud

## How to install
* Make sure your `.env.local` file doesn't have a `REACT_APP_ELASTIC_URL` or `REACT_APP_PROXY_URL` pointing to production or test sites. If these are missing the search should correctly use the local ES index. 
* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-10016-sonarcloud-fixes`
  * `make fresh`
* Run `make drush-cr`
* If you have tens of thousands of decision nodes from production, consider deleting all of those entities with `drush edel node --bundle=decision`  to make indexing them faster. You can also use this command: https://github.com/City-of-Helsinki/helsinki-paatokset/blob/dev/public/modules/custom/paatokset_search/src/Commands/DevelopmentDatabaseCleanerCommand.php

## How to test
* Aggregation and data fetching commands should work the same as before:
  * `drush ap:agg meetings --start=2024-10-01T00:00:00 --end=2025-01-01T00:00:00 --queue -v`
  * `drush ap:agg cases --start=2024-10-01T00:00:00 --end=2025-01-01T00:00:00 --queue -v`
  * `druah ap:agg decisions --start=2024-11-15T00:00:00 --end=2024-12-05T00:00:00 --queue -v` 
  *  [x] All of the above should pull in some data into the queue
  * `drush queue:run ahjo_api_aggregation_queue -v; drush ap:gm -v; drush ap:ud -v`
  * [x] The queue should process normally, followed by motion generation based on meeting data, then the decision data should be fetched correctly. Missing cases and meetings should be added to the queue.
  * `drush queue:run ahjo_api_aggregation_queue -v; drush ap:gm -v;drush ap:ud --logic=case -v`
  *  [x] This should fetch the missing data.
* Indexing content should work correctly:
  * `drush sapi-c; drush sapi-rt; drush sapi -i` (This might take a very long time if you have lots of content)
  * [x] Check that the decisions search app works normally: https://helsinki-paatokset.docker.so/fi/asia
  * [x] Check that the policymaker search works normally: https://helsinki-paatokset.docker.so/fi/paattajat
  * [x] Check that the meetings calendar works normally: https://helsinki-paatokset.docker.so/fi/kokouskalenteri
* Decision navigation and dropdowns:
  * [x] Check a few cases with multiple decisions bundled into it. The arrow and dropdown navigations should work normally.
* [x] Tabbed list navigation should work normally: https://helsinki-paatokset.docker.so/fi/paattajat/kaupunginvaltuusto/asiakirjat

[UHF-10016]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-10016?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
